### PR TITLE
Add tiny introduction to dictionaries to core-concepts

### PR DESCRIPTION
**Pull request by lindsay-fredrick**
This is a super tiny update to core-concepts to get dictionaries in there. Sorry @ateucher but it's gotta happen!

### CODE REVIEW

File: course-content/02_core_data_structure_concepts.qmd [hunk 1]
```diff
@@ -230,6 +234,52 @@ What would these expressions return?
...
+```{python}
+#| error: true
+my_dict[0:5]
+```
+
+Dictionaries are an abstract data type that can take a while to get used to! They can be a powerful tool in Python. Common uses for dictionaries include:
+
+ * Creating searchable parameter lists for models
+ * Supplying extra arguments to functions 
+ * Storing complex outputs or datasets
```
Comment (ateucher) [inline course-content/02_core_data_structure_concepts.qmd:279]: Are lookup tables also a common use?

File: course-content/02_core_data_structure_concepts.qmd [hunk 2]
```diff
@@ -230,6 +234,52 @@ What would these expressions return?
...
+
+Another object that exists in Python is the `dictionary`. It is similar to a list in that it can hold a variety of different types of objects inside of it. However an important difference is in how we access these objects. With a list (or string), we have an ordered arrangement of items that we access with an **integer index**. However, we access the **values** in a dictionary with a **key**, which can be anything we want.
+
+Let's build a dictionary, which is denoted in Python with curly `{}` brackets:
+
+```{python}
+my_dict = {
+    'first_key': 'some value',
+    'A': ['a', 'differerent', 'type', 'of', 'object'],
+    2: False
```
Comment (ateucher) [inline course-content/02_core_data_structure_concepts.qmd:247]: Interesting that the keys don't have to be same type (i.e., a string). Can you / do you ever look up more than one element of a dict?

**Review by ateucher — APPROVED** [review]
A couple of questions/comments, but looks good!

File: course-content/02_core_data_structure_concepts.qmd (hunk 1 above)
Comment (lindsay-fredrick) [reply course-content/02_core_data_structure_concepts.qmd:279]: Yeah you could say that, it basically is just a fancy lookup table.

File: course-content/02_core_data_structure_concepts.qmd (hunk 2 above)
Comment (lindsay-fredrick) [reply course-content/02_core_data_structure_concepts.qmd:247]: There's no specifically built in way to do it (you can't do my_dict['first_key', 'A']  or anything). It is doable, but more complicated than worth mentioning here I'd say.

**Pull request merged**